packages/boxel-ui/addon/src/components/select/trigger.gts

```diff
@@ -0,0 +1,114 @@
+import Component from '@glimmer/component';
+import type { Select } from 'ember-power-select/components/power-select';
+
+import CaretDown from '../../icons/caret-down.gts';
+import CaretUp from '../../icons/caret-up.gts';
+
+export interface TriggerSignature {
+  Args: {
+    placeholder?: string;
+    select: Select;
+    selectedItemComponent?: any;
+  };
+  Blocks: {
+    default: [Select['selected'], Select];
+    icon: [];
+    placeholder: [];
+  };
+  Element: HTMLElement;
+}
+
+// This component is used to provide a consistent boxel style to trigger components
+// Abstain from using this component with many data customizations
+export class BoxelTriggerWrapper extends Component<TriggerSignature> {
+  <template>
+    <div class='boxel-trigger'>
+      <div class='boxel-trigger-content'>
+        {{#if (has-block 'placeholder')}}
+          <div class='boxel-trigger-placeholder'>
+            {{yield to='placeholder'}}
+          </div>
+        {{/if}}
+        {{! Yield the selected item and the select component to allow ember power select consumers to specify the selected item to be displayed after selection}}
+        {{! This is how ember-power-select does it inside of their default component }}
+        {{yield @select.selected @select}}
```

Comment:
I would change it this way:
```
        {{#if @select.selected}}
          {{yield @select.selected @select}}
        {{else if (has-block 'placeholder')}}
          <div class='boxel-trigger-placeholder'>
            {{yield to='placeholder'}}
          </div>
        {{/if}}
```

This way you wouldn't need the `this.showPlaceholder` getter in the DefaultTrigger class below.

Reply:
I move showPlaceholder into the wrapper and removed the usage of blocks. I don't know why there was both blocks and args of placeholder. 

It looks like I want showPlaceholder bcos it may handle a case for select or multi-select

ie @select.selected = undefined for select and @select.select = [] for multi-select